purchase_request/wizard/purchase_request_line_make_purchase_order.py

```diff
@@ -200,8 +200,16 @@ def _get_order_line_search_domain(self, order, item):
             ("name", "=", name),
             ("product_id", "=", item.product_id.id),
             ("product_uom", "=", vals["product_uom"]),
-            ("analytic_distribution", "=?", item.line_id.analytic_distribution),
         ]
+
+        if item.line_id.analytic_distribution:
+            analytic_account_ids = list(item.line_id.analytic_distribution.keys())
+            order_line_data.append(
+                ("analytic_distribution", "in", analytic_account_ids)
+            )
+        else:
+            order_line_data.append(("analytic_distribution", "=", False))
```

Comment:
The `=?` operator evaluates as True if the right hand side is falsy. That means that a request line without analytics can be linked to any order line with or without analytics. Did you mean to change that behaviour?

Reply:
The `=?` operator doesn't support when searching the field `analytic_distribution` anymore.

```
    def _condition_to_sql(self, alias: str, fname: str, operator: str, value, query: Query) -> SQL:
        if fname != 'analytic_distribution' or self.env.context.get('account_report_analytic_groupby'):
            return super()._condition_to_sql(alias, fname, operator, value, query)

        if operator not in ('=', '!=', 'ilike', 'not ilike', 'in', 'not in'):
            raise UserError(_('Operation not supported'))
```

`odoo/addons/analytic/models/analytic_mixin.py`

Reply:
Yes indeed! Do you agree that we should still keep the original behaviour, which means ignoring the analytics of existing purchase lines in case of a request line that has no analytics set?

Reply:
The `=?` is unset or equal to, which means including the analytics of existing purchase lines has no analytics set.

I am keeping the original behavior.

Reply:
I'm sorry, you are absolutely right!